### ISSUE TITLE
Strengthen test on replicate IDs in survCheckData.

### ISSUE THREAD

**Issue by pveber** [pveber/morse#14]
See my [comment in the code](https://github.com/pveber/mosaic-r/blob/master/R/survDataCheck.R#L205). I think for a given concentration, we should have the same replicate labels for all time points.

**Comment by philipperuiz** [comment]
We can add this test, it checks if, for each time and concentration the label of the replicates are the same as the control : 
 # 11 test label of replicate for each time and concentration
  reslab <- by(data, list(data$conc, data$time),
               function(x) {str_c(x$replicate, collapse = "")}) # combine replicate label

  replicatelabel <- str_c(unique(subset(data, data$conc == min(data$conc) & data$time == 0)$replicate), collapse = "") # reference

  if (!any(reslab) == replicatelabel)  { # test
    err <- error("ReplicateLabel",
                 "For at least one time and one concentration a replicate label is differente from the others.")
    errors <- rbind(errors, err)
  }

**Comment by pveber** [comment]
Not bad! I think there's still a problem though: this assumes that inside a (concentration, time), replicates are always sorted the same way. I think you should add a sort, like in

```R
reslab <- by(data,
             list(data$conc, data$time),
             function(x) {str_c(sort(x$replicate), collapse = "")})
```

Also, the definition of `replicatelabel` seems useless to me, the test could simply be:
```R
if (any(reslab != reslab[[1]])) {
  ...
}
```
BTW, note your parenthesis for the call to `any` are wrong. Could you provide a PR for that?

**Comment by pveber** [comment]
fixed in febbbfb9, closing.
